src/otx/cli/test.py

```diff
@@ -47,6 +29,27 @@ def otx_test(overrides: list[str]) -> None:
         configure_hydra_outputs(cfg)
 
         # test the model
-        from otx.core.engine.test import test
-
-        metric_dict, _ = test(cfg)
+        from otx.core.data.module import OTXDataModule
+
+        log.info(f"Instantiating datamodule <{cfg.data}>")
+        datamodule = OTXDataModule(task=cfg.base.task, config=cfg.data)
+
+        log.info(f"Instantiating model <{cfg.model}>")
+        model: OTXModel = hydra.utils.instantiate(cfg.model.otx_model)
+        optimizer = hydra.utils.instantiate(cfg.model.optimizer)
+        scheduler = hydra.utils.instantiate(cfg.model.scheduler)
+
+        from otx.engine import Engine
+
+        trainer_kwargs = {**cfg.trainer}
+        engine = Engine(
+            task=cfg.base.task,
+            work_dir=cfg.base.output_dir,
+            model=model,
+            optimizer=optimizer,
+            scheduler=scheduler,
+            datamodule=datamodule,
+            checkpoint=cfg.checkpoint,
+            device=trainer_kwargs.pop("accelerator", "auto"),
```

Comment:
Is there any hypothetical plan how to avoid this workaround?
I expect growing number of implicit configuration parsing and even editing when more features from 1.x would be restored in 2.0

Reply:
@sovrasov Hi, This cli file will be refactored. You can see the future CLI in this PR.
https://github.com/openvinotoolkit/training_extensions/pull/2772

The current CLI is temporary code.

Reply:
I think your concerns can be addressed later via the CLI.